src/mame/drivers/meritum.cpp

```diff
@@ -91,30 +118,35 @@ class meritum_state : public driver_device
 	required_device<speaker_sound_device> m_speaker;
 	required_device<cassette_image_device> m_cassette;
 	required_ioport_array<8> m_io_keyboard;
+	optional_device<centronics_device> m_centronics;
+	required_device<input_merger_device> m_nmigate;
 };
 
 void meritum_state::mem_map(address_map &map)
 {
 	map(0x0000, 0x37ff).rom();
 	map(0x3800, 0x38ff).mirror(0x300).r(FUNC(meritum_state::keyboard_r));
 	map(0x3c00, 0x3fff).ram().share(m_p_videoram);
-	map(0x4000, 0xffff).ram();
+	// Model 1 got 16 (or 17) KiB of RAM, Model 2: 32 or 48 KiB
+	map(0x4000, 1 == m_model ? 0x7fff /*16 KiB*/: 0xffff /*48 KiB*/).ram();
```

Comment:
Please use separate memory map functions for the different memory layouts – switching on variables like this makes it harder to follow.